### ISSUE TITLE
OS.File.write fails on Windows with permission denied

### ISSUE THREAD

**Issue by hcarty** [dbuenzli/bos#59]
Example code:
```ocaml
open Rresult
open Bos

let () =
  match OS.File.write (Fpath.v "w.out") "content" with
  | Ok () -> ()
  | Error msg -> print_endline (Fmt.strf "Error: %a" R.pp_msg msg)
```
When runs it prints `Error: rename .\bos-701c75.tmp to w.out: Permission denied`.

**Comment by hcarty** [comment]
Perhaps this is caused by renaming the file while it's still open?  Opening with `O_SHARE_DELETE` may help in this case.

**Comment by dbuenzli** [comment]
Fixed by @hcarty in  b4143e089d5bfb4b46c08
